azurerm/resource_arm_data_factory_linked_service_sql_server.go

```diff
@@ -0,0 +1,357 @@
+	sort.Strings(newSplit)
+
+	// We need to remove the password from the new string since it isn't returned from the api
+	for i, v := range newSplit {
+		if strings.HasPrefix(v, "password") {
+			newSplit = append(newSplit[:i], newSplit[i+1:]...)
+		}
+	}
+
+	if len(oldSplit) != len(newSplit) {
+		return false
+	}
+
+	// We'll error out if we find any differences between the old and the new connectiong strings
+	for i := range oldSplit {
+		if !strings.EqualFold(oldSplit[i], newSplit[i]) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func validateAzureRMDataFactoryLinkedServiceName(v interface{}, k string) (warnings []string, errors []error) {
+	value := v.(string)
+	if regexp.MustCompile(`^[-.+?/<>*%&:\\]+$`).MatchString(value) {
+		errors = append(errors, fmt.Errorf("any of '-' '.', '+', '?', '/', '<', '>', '*', '%%', '&', ':', '\\', are not allowed in %q: %q", k, value))
+	}
+
+	return warnings, errors
+}
+
+func expandDataFactoryLinkedServiceIntegrationRuntime(d *schema.ResourceData) *datafactory.IntegrationRuntimeReference {
+	if v, ok := d.GetOk("integration_runtime_name"); ok {
+		integrationRuntimeName := v.(string)
+		typeString := "IntegrationRuntimeReference"
+
+		return &datafactory.IntegrationRuntimeReference{
+			ReferenceName: &integrationRuntimeName,
+			Type:          &typeString,
+		}
+	}
+	return nil
+}
+
+func expandDataFactoryLinkedServiceParameters(d *schema.ResourceData) map[string]*datafactory.ParameterSpecification {
+	input := d.Get("parameters").(map[string]interface{})
+
+	output := make(map[string]*datafactory.ParameterSpecification)
+
+	for k, v := range input {
+		output[k] = &datafactory.ParameterSpecification{
+			Type:         datafactory.ParameterTypeString,
+			DefaultValue: v.(string),
+		}
+	}
+
+	return output
+}
+
+func expandDataFactoryLinkedServiceAdditionalProperties(d *schema.ResourceData) map[string]interface{} {
+	input := d.Get("additional_properties").(map[string]interface{})
+
+	output := make(map[string]interface{})
+
+	for k, v := range input {
+		output[k] = v
+	}
+
+	return output
+}
+
+func flattenDataFactoryLinkedServiceParameters(input map[string]*datafactory.ParameterSpecification) map[string]interface{} {
+	output := make(map[string]interface{})
+
+	for k, v := range input {
+		if v != nil {
+			// we only support string parameters at this time
+			val, ok := v.DefaultValue.(string)
+			if !ok {
+				log.Printf("[DEBUG] Skipping parameter %q since it's not a string", k)
+			}
+
+			output[k] = val
+		}
+	}
+
+	return output
+}
+
+func expandDataFactoryLinkedServiceAnnotations(d *schema.ResourceData) *[]interface{} {
+	input := d.Get("annotations").([]interface{})
+	annotations := make([]interface{}, 0)
+
+	for _, annotation := range input {
+		annotations = append(annotations, annotation.(string))
+	}
+
+	return &annotations
+}
+
+func flattenDataFactoryLinkedServiceAnnotations(input *[]interface{}) []string {
+	annotations := make([]string, 0)
+	if input != nil {
```

Comment:
rather than nesting this - we'd generally inverse this:

```suggestion
	if input == nil {
	  return annotations
	}
```